### DISCUSSION TITLE
Översättning av nyckelbegrepp

### DISCUSSION THREAD

**Discussion by chrillof** [progit2-sv/progit2#4] [General]
Här diskuterar vi översättningen av specifika begrepp så att det skall bli så enhetligt och lättförståeligt som möjligt.

**Comment by evelinabe** [comment]
Har stött på _feature branch_. Ser att Azure översatt _feature branch_ till _funktionsgren_. Det låter... tungt. I Computer Swedens språkordlista föreslås _funktion, egenskap, finess_. _Egenskapsgren eller finessgren_ låter också konstigt. Gillade däremot: "är det en bugg eller en finess". 

Exempelmening från `book/05-distributed-git/sections/contributing-asc`:

**"You'll learn how to work in an environment where small groups collaborate on features".**
Du lär dig hur man arbetar en miljö där mindre grupper samarbetar för att ta fram funktioner/egenskaper/finesser.

**"Let's follow Jessica's workflow as she works on her two features."**
Vi följer med Jessica när hon arbetar på sina två funktioner/egenskaper/finesser.

**"She creates a new branch for the feature and does some work on it."**
Hon gör en ny gren för funktionen och arbetar med den.

**Reply by chrillof** [reply]
Jag skulle nog översätta feature till funktion och feature branch till funktionsgren. Det är ju ofta så man jobbar vid nyutveckling, tänker jag. Men man kan ju, där det är lämpligt av kontexten, använda egenskap eller finess.

**Reply by evelinabe** [reply]
Tycker inte att funktion och feature egentligen är samma sak. Funktion kan ju vara lite vad som helst medan en feature är en karaktäristisk egenskap, något som utmärker sig utöver det grundläggande.

**Reply by chrillof** [reply]
Nä, funktion och feature mappar inte 1:1, men man kanske får använda lite fler begrepp beroende på kontexten som sagt. I regel pratar man väl om _funktionsutveckling_ snarare än _egenskapsutveckling_ eller _finessutveckling_. Det låter åtminstone bäst av de tre i mina öron.

**Reply by nafmo** [reply]
I översättningen av git/gitk/git-gui så använder jag "funktion" för "feature", det är den översättning som i mina öron "låter" bäst och passar bäst in i texten, så "funktionsgren" tycker jag fungerar fint. Jag använder nog också "funktioner" för "features" också i vanlig löptext.

**Comment by evelinabe** [comment]
**Fork = klyva/förgrena/dela/gaffla/kopiera...?**

Azure pratar om [gafflar](https://learn.microsoft.com/sv-se/azure/devops/repos/git/forks?view=azure-devops&tabs=visual-studio).
Jag förstår inte den associationen, och har väldigt svårt att ta till mig det. Har noterat att det används mer internationellt, t.ex. appen trail forks, och känner till stämgaffel och framgaffel, men för mig har de det gemensamt att själva förgreningarna ser likadana ut. Och även om alla forkade repon har samma källa så är de ju inte styrda att sen se ut på samma sätt. 

Det som jag förstår att man vill komma till är väl just det att det forkade repot inte är helt och hållet avhugget från ursprungskällan. Jag måste bråttas lite med det här 😆


Exempelmening: 
This first example describes contributing via forking.
I det första exemplet går vi igenom hur man bidrar genom att gaffla.

**Reply by chrillof** [reply]
Man får passa sig här så att man inte förväxlar det med vanlig branching (förgrena). Två "forkade" arkiv har ju inte samma källa utan de är faktiska kopior och har inget med varandra att göra. Det de har gemensamt är ju att de har delvis identisk historik som man kan utnyttja. Detsamma när man har t.ex. två olika fjärrarkiv. Man skulle kunna använda begreppet "klon", eftersom ett klonat repo de facto är en fork av ett annat repo (fast som i regel även har en arbetsyta). Men, det kan också bli rörigt när man pratar om processen att klona ett arkiv.

**Reply by chrillof** [reply]
Man kanske rent av skall säga just "kopia"?

**Reply by evelinabe** [reply]
Ja. Det behöver ju vara tydligt på något sätt att branching, clone och fork är tre olika saker.  

Så förgrening, klon och kopia...?

**Reply by evelinabe** [reply]
Eller hm... En fork är ju helt fristående. När jag gör uppdateringar till open source här på github så måste jag ju då och då göra en pull från originalrepot till mitt forkade repo för att få in deras senaste ändringar så jag inte skickar in PR:s på gammal kod. Så till skillnad från en kopia hänger ju forken kvar. Skillnaden är att när man klonar har man skrivbehörighet till den källan, när man forkar har man skrivbehörighet till sin kopia men inte till ursprungsrepot. Det är fristående, ja, men inte avklippt.

**Reply by chrillof** [reply]
I min värld är fork bara en ren kopia av ett annat arkiv, precis som när man gör `git clone`. Det finns således inget "originalarkiv" att tala om eftersom alla arkiv är fullständiga kopior (undantaget medvetet avhuggna kloner, dvs "_shallow clone_"). Att man sedan kan hämta uppdateringar från ett annat repo är precis som att lokalt göra `git fetch` från valfritt fjärrarkiv. Varje arkiv lever dock självständigt även om man kan synka mellan dem.

Den enda kopplingen mellan en fork och dess "originalarkiv" är att "forken" har konfigurerat "originalarkivet" som ett fjärrarkiv. En fork och dess "originalarkiv" kan divergera på samma sätt som ditt lokala gitarkiv och ett fjärrarkiv. "Originalarkivet" kan tas bort från jordens yta och dess "forkar" kommer inte påverkas, nämnvärt. Man kan i teorin även lägga in kopplingar mellan flera "forkar" av samma arkiv (dock inget jag provat, men det borde funka 😄). Är du med på hur jag tänker, eller missförstår jag dig?

Det känns som att "fork" är som en branch på en högre nivån, en branch av ett arkiv snarare än en branch i arkivet. Svårt att hitta begrepp som inte förvirrar folk. Att det är en kopia gäller ju bara i samma ögonblick som "forken" skapas, när de väl divergerar blir ju arkiven delmängder av varandra. 🤔

**Reply by chrillof** [reply]
Knoppa? 🤔

**Reply by nafmo** [reply]
Den här är lite knepig. En "fork" (substantiv) är ju en gaffel, men "fork" (verb) är definitivt inte någon gaffel. En förgrening eller gren passar här (tänk en väg som delas i två), men man riskerar att krocka med "branch" som också är "gren". I översättningen av git/git-gui/gitk använder jag huvudsakligen "gren", men förtydligar med "(fork)" om det kan missförstås.

**Reply by evelinabe** [reply]
Det svenska verbet "gaffla" är inte heller en gaffel. Synd att det inte har något med förgreningar att göra bara, det hade gjort det hela mycket enklare.

**Comment by evelinabe** [comment]
**Push vs send**

En sak som stör mig lite med använda skicka är att det liksom inte beskriver att det man skickar faktiskt trycks in i det befintliga. När man skickar något vet man inte hur det tas emot, det kan hamna i en kö någonstans eller hamna vid sidan av. Här trycker man ju in sina ändringar i den existerande versionen oavsett vad en mottagare skulle kunna tycka (det är därför den behöver ha tyckt innan)

**Reply by chrillof** [reply]
Hmm, jag är med på hur du tänker. Man kanske skall säga _delge_, men det låter kanske också lite för högtravande? Det var väl _sända_ man använde i den officiella översättningen, då bör vi kanske inte hitta på ett eget begrepp här för att undvika förvirring.

**Reply by nafmo** [reply]
Ge mig gärna besked om ni hittar en bra översättning. I git och git-gui är "push" översatt till "sänd" i brist på bättre ord. Eftersom det inte finns någon "send" så är risken för förväxlingar minimal.
